### PR TITLE
fix(core): recurse input of workflows when using toil provider

### PR DESCRIPTION
**Pull request by vigsterkr**
# Description

This fixes the execution of a workflow using toil, when the input of a workflow is a directory that contains directories as children.

### CODE REVIEW

File: renku/core/management/workflow/providers/toil.py [hunk 1]
```diff
@@ -77,19 +77,23 @@ def run(self, storage):
...
+            input_path = Path(input)
+
             if isinstance(file_metadata, dict):
-                input_path.mkdir(exist_ok=True)
+                input_path.mkdir(parents=True, exist_ok=True)
                 for path, file_id in file_metadata.items():
-                    storage.readGlobalFile(file_id, userPath=path)
-            else:
+                    _read_input(path, file_id)
+            elif not input_path.exists():
```
Comment (mohammad-sdsc) [inline renku/core/management/workflow/providers/toil.py:87]: This won't `readGlobalFile` if the `input_path` exists; is this the correct behavior?

**Review by mohammad-sdsc — COMMENTED** [review]
LGTM, just made two minor comments.

File: renku/core/management/workflow/providers/toil.py (hunk 1 above)
Comment (vigsterkr) [reply renku/core/management/workflow/providers/toil.py:87]: yes that's intentional. `readGlobalFile` has no option for overwrite, i.e. it throws an exception if `userPath` already exists; this avoid pulling the given file from the global storage if it's already available.

**Review by mohammad-sdsc — APPROVED** [review]
Thank you!

**Pull request merged**